collections/Quests/NumberInput/EditView.tsx

```diff
@@ -0,0 +1,90 @@
+import { useForm } from 'react-hook-form'
+import { InputField } from '@/components/Elements/FormElements'
+import { useQuestStore } from './store'
+import { valNumberInput } from '@/collections/Quests/NumberInput/validation'
+import { yupResolver } from '@hookform/resolvers/yup'
+import { SelectField } from '@/components/Elements/Select/SelectField'
+import { Units } from '@/collections/Quests/NumberInput/units'
+import { SelectOption } from '@/components/Elements/Select'
+
+const unitOptions: SelectOption[] = [
+  { value: Units.None },
+  { value: Units.Meter },
+  { value: Units.Millimeter },
+  { value: Units.Centimeter },
+  { value: Units.Decimeter },
+  { value: Units.Kilometer },
+  { value: Units.Squaremeter },
+  { value: Units.Squarekilometer },
+  { value: Units.Cubicmeter },
+  { value: Units.Litre },
+  { value: Units.Millilitre },
+  { value: Units.Centilitre },
+  { value: Units.Decilitre },
+  { value: Units.Hectolitre },
+  { value: Units.Kilogram },
+  { value: Units.Milligram },
+  { value: Units.Gram },
+  { value: Units.Pound },
+  { value: Units.Radian },
+  { value: Units.Degree },
+  { value: Units.Second },
+  { value: Units.Minute },
+  { value: Units.Hour },
+  { value: Units.Day },
+  { value: Units.Percent },
+]
```

Comment:
this is not good if there are more units in the future, change to:
```suggestion
const unitOptions: SelectOption[] = Object.values(Units).map(unit => ({value: unit}))
```